wallet/wallet.h

```diff
@@ -1246,6 +1246,20 @@ void add_unreleased_tx(struct wallet *w, struct unreleased_tx *utx);
 /* These will touch the db, so need to be explicitly freed. */
 void free_unreleased_txs(struct wallet *w);
 
+/* wallet_persist_utxo_reservation - Removes destructor
+ *
+ * Persists the reservation in the database (until a restart)
+ * instead of clearing the reservation when the utxo object
+ * is destroyed */
```

Comment:
I think this is the code you're looking for https://github.com/ElementsProject/lightning/blob/master/wallet/wallet.c#L3723-L3758

Reply:
I'm pulling out the 'reservation system' from #3418 right now, which will make reservations persistent/expirable.

Reply:
That's an excellent idea, what timeout are you looking at for default? (pretty hesitant to add yet more arguments to commands, since 90% are simply unused)

Reply:
Also, good work on finding the un-reserve on startup, I was getting lost looking for it :wink:

Reply:
Ok so this is now available in #3789. I defaulted to 144 blocks, or 24 hours. You can disable it or choose a different number, but that'll revert to the current behavior of resetting the utxo reservation if the node is restarted.

One nice feature of the way we 'broadcast' txs is that if you send a tx via c-lightning it will mark it as 'spent' (really this means 'in mempool'), so attempting to spend a tx will prevent a tx from being re-used/expired.